ui/lib/ldap/addon/components/page/configuration.ts

```diff
@@ -0,0 +1,82 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
+import errorMessage from 'vault/utils/error-message';
+
+import type LdapConfigModel from 'vault/models/ldap/config';
+import type SecretEngineModel from 'vault/models/secret-engine';
+import type AdapterError from 'ember-data/adapter'; // eslint-disable-line ember/use-ember-data-rfc-395-imports
+import type { Breadcrumb } from 'vault/vault/app-types';
+import type FlashMessageService from 'vault/services/flash-messages';
+
+interface Args {
+  configModel: LdapConfigModel;
+  configError: AdapterError;
+  backendModel: SecretEngineModel;
+  breadcrumbs: Array<Breadcrumb>;
+}
+
+interface Field {
+  label: string;
+  value: any; // eslint-disable-line @typescript-eslint/no-explicit-any
+  formatTtl?: boolean;
+}
+
+export default class LdapConfigurationPageComponent extends Component<Args> {
+  @service declare readonly flashMessages: FlashMessageService;
+
+  get defaultFields(): Array<Field> {
+    const model = this.args.configModel;
+    const keys = [
+      'binddn',
+      'url',
+      'schema',
+      'password_policy',
+      'userdn',
+      'userattr',
+      'connection_timeout',
+      'request_timeout',
+    ];
+    return model.allFields.reduce<Array<Field>>((filtered, field) => {
```

Comment:
Note for future: I'm thinking about combining the two model decorators, which would add `allByKey` to the model and would make all of this a lot simpler (just map keys and pick from `allByKey` by the name)

Reply:
Oh yes I like that idea! Seems like it would be easier to work with than an array in most instances and could still be iterated over in the template with `each-in`!